### ISSUE TITLE
xmlToRuby NullPointerException

### ISSUE THREAD

**Issue by boskowski** [jruby/activerecord-jdbc-adapter#615]
When reading a record from a PostgreSQL table with an XML data type, `arjdbc.jdbc.RubyJdbcConnection.xmlToRuby` consistently throws a Java NullPointerException if the XML column holds a NULL value.

Here an example:
```
>> XmlTable.create(name: 'one', xml: '<foo>bar</foo>')
  SQL (2.0ms)  INSERT INTO "xml_tables" ("name", "xml") VALUES ('one', xml '<foo>bar</foo>') RETURNING "uuid"
#<XmlTable uuid: "b87fdcb2-9794-4f52-8113-196156dc1745", name: "one", xml: "<foo>bar</foo>">

>> r=XmlTable.first
  XmlTable Load (2.0ms)  SELECT  "xml_tables".* FROM "xml_tables"   ORDER BY "xml_tables"."uuid" ASC LIMIT 1
#<XmlTable uuid: "b87fdcb2-9794-4f52-8113-196156dc1745", name: "one", xml: "<foo>bar</foo>">

>> r.xml=nil
nil

>> r.save
  SQL (1.0ms)  UPDATE "xml_tables" SET "xml" = NULL WHERE "xml_tables"."uuid" = 'b87fdcb2-9794-4f52-8113-196156dc1745'
true

>> r.reload
  XmlTable Load (2.0ms)  SELECT  "xml_tables".* FROM "xml_tables"  WHERE "xml_tables"."uuid" = 'b87fdcb2-9794-4f52-8113-196156dc1745' LIMIT 1
Java::JavaLang::NullPointerException: : SELECT  "xml_tables".* FROM "xml_tables"  WHERE "xml_tables"."uuid" = 'b87fdcb2-9794-4f52-8113-196156dc1745' LIMIT 1
Java::JavaLang::NullPointerException: 
	from arjdbc.jdbc.RubyJdbcConnection.xmlToRuby(RubyJdbcConnection.java:1964)
	from arjdbc.jdbc.RubyJdbcConnection.jdbcToRuby(RubyJdbcConnection.java:1618)
	from arjdbc.postgresql.PostgreSQLRubyJdbcConnection.jdbcToRuby(PostgreSQLRubyJdbcConnection.java:478)
	from arjdbc.jdbc.RubyJdbcConnection$ResultHandler.mapRow(RubyJdbcConnection.java:3412)
	from arjdbc.jdbc.RubyJdbcConnection.mapToResult(RubyJdbcConnection.java:1559)
	from arjdbc.jdbc.RubyJdbcConnection.mapQueryResult(RubyJdbcConnection.java:949)
	from arjdbc.jdbc.RubyJdbcConnection.access$200(RubyJdbcConnection.java:94)
	from arjdbc.jdbc.RubyJdbcConnection$9.call(RubyJdbcConnection.java:913)
	from arjdbc.jdbc.RubyJdbcConnection$9.call(RubyJdbcConnection.java:906)
	from arjdbc.jdbc.RubyJdbcConnection.withConnection(RubyJdbcConnection.java:3209)
	from arjdbc.jdbc.RubyJdbcConnection.withConnection(RubyJdbcConnection.java:3190)
	from arjdbc.jdbc.RubyJdbcConnection.executeQuery(RubyJdbcConnection.java:906)
	from arjdbc.jdbc.RubyJdbcConnection.execute_query(RubyJdbcConnection.java:847)
	from arjdbc.jdbc.RubyJdbcConnection$INVOKER$i$execute_query.call(RubyJdbcConnection$INVOKER$i$execute_query.gen)
	from org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:168)
	from org.jruby.ast.CallOneArgNode.interpret(CallOneArgNode.java:57)
... 427 levels...
	from org.jruby.RubyKernel.load19(RubyKernel.java:1087)
	from org.jruby.RubyKernel$INVOKER$s$0$1$load19.call(RubyKernel$INVOKER$s$0$1$load19.gen)
	from org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:210)
	from org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:206)
	from org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:326)
	from org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:170)
	from ruby.__dash_e__.__file__(-e:1)
	from ruby.__dash_e__.load(-e)
	from org.jruby.Ruby.runScript(Ruby.java:866)
	from org.jruby.Ruby.runScript(Ruby.java:859)
	from org.jruby.Ruby.runNormally(Ruby.java:728)
	from org.jruby.Ruby.runFromMain(Ruby.java:577)
	from org.jruby.Main.doRunFromMain(Main.java:395)
	from org.jruby.Main.internalRun(Main.java:290)
	from org.jruby.Main.run(Main.java:217)
	from org.jruby.Main.main(Main.java:197)
```

Versions used:
* jruby 1.7.18 (jruby 1.7.18 (1.9.3p551) 2014-12-22 625381c on Java HotSpot(TM) 64-Bit Server VM 1.8.0_31-b13 +jit [Windows 7-amd64])
* activerecord 4.1.8
* activerecord-jdbc-adapter 1.3.13
* activerecord-jdbcpostgresql-adapter 1.3.13
* jdbc-postgres 9.3.1102

**Comment by kares** [comment]
@boskowski thanks a lot, this is an ugly bug - you can count on the fix being in next release (**1.3.14**)

**Comment by boskowski** [comment]
Thank you very much!
